content/reference/custom_interactions.md

```diff
@@ -0,0 +1,221 @@
+modal (basically a pop-up form). Within custom commands it is possible to both create and customize these new
+interactive features, but respond to them as well, opening up new possibilities for ephemeral message responses, modals,
+and more within YAGPDB custom templating script.
+
+### Definitions
+
+**Interaction** - A user engaging with YAGPDB through one of Discord's builtin features: Clicking a button, Making a
+selection with a select menu, or Submitting a modal.
+
+**Response** - YAGPDB is required to send an initial interaction response after receiving an *interaction*. If it does
+not do this, the user triggering the interaction will see a "This application did not respond" error. The bot cannot
+respond to an interaction more than once. YAGPDB may only respond to an interaction in one of the following ways:
+Sending a message, sending a modal, or updating the message which the *interaction* was triggered on.
+
+**Followup** - Since YAGPDB may only *respond* to an *interaction* once, it is subsequently required to send an interaction
+followup if it still needs to respond to the interaction. These followups can be sent up to 15 minutes after the initial
+interaction, and you can send as many as you want. YAGPDB may only send a followup in one of the following ways: Sending
+a message, editing a response or followup, or getting a response or followup message.
+
+**Interactive Elements** - Elements users can interact with to send *interactions*, i.e. buttons, menus, and modals.
+
+**Message Components** - *Interactive Elements* which can be attached to YAGPDB's Discord messages, i.e. buttons and menus.
+
+**Button** - A button appearing in or under a Discord message sent by YAGPDB. You can create and customize these
+buttons' appearance and behavior with color, emoji, label text, etc. When a button is clicked, an *interaction* is sent
+to the bot.
+
+**Menu** - A dropdown select menu appearing in or under a Discord message sent by YAGPDB. You can create and customize these
+menus' appearance and behavior with placeholder text, predefined options with labels, descriptions, and/or emojis,
+designate the entire menu as a user or role select menu instead, etc. When a select menu is used, an *interaction* is sent
+to the bot.
+
+**Modal** - A pop-up form YAGPDB can send in response to an interaction. It allows users to privately input text which
+is sent directly to YAGPDB for use in CC scripting. You can create and customize these modals' appearance and
+behavior with a title and fields. YAGPDB can both **receive a submitted modal** (which is an
+*interaction*), and **send a modal** for a member to fill out, (which is an interaction *response*).
+
+**Ephemeral** - An ephemeral message is sent to a server channel but only appears to a single user. YAGPDB cannot send
+these ephemeral messages to users except in response to an *interaction*. Both *response* messages and *followup*
+messages can be ephemeral.
+
+## Interaction Lifetime
```

Comment:
This is a nice summary (with a visual as well!) I would suggest it be placed before the definition as a sort of "basic intro". In general I believe guides or tutorials (which this article appears to be, as opposed to reference documentation per se) should begin with a brief overview that still has some usable technical detail before diving into specific definitions and so on.